cdc/sink/flowcontrol/flow_control.go

```diff
@@ -169,17 +160,16 @@ func (c *TableFlowController) enqueueSingleMsg(
 	c.queueMu.Lock()
 	defer c.queueMu.Unlock()
 
-	e := c.queueMu.queue.Back()
 	// 1. Processing a new txn with different commitTs.
-	if e == nil || lastCommitTs < commitTs {
+	if c.queue.Empty() || lastCommitTs < commitTs {
 		atomic.StoreUint64(&c.lastCommitTs, commitTs)
 		c.resetBatch(lastCommitTs, commitTs)
 		c.addEntry(msg, size)
 		return
 	}
 
 	// Processing txns with the same commitTs.
-	txnEntry := e.(*txnSizeEntry)
+	txnEntry, _ := c.queue.Tail()
```

Comment:
ditto

Reply:
Here we've checked the queue is not empty(line 164), while the second return value is false only when the queue is empty.